src/README.md

```diff
@@ -1,8 +1,296 @@
-## Documentation
+# Understanding API-Flow
+This document tries to briefly explain how API-Flow functions and how you can extend it.
 
-API-Flow can be decomposed into three main components:
+## The API-Flow conversion flow
+API-Flow converts an API format into another in 4 different steps
+1. **Setup -** API-Flow sets the environment in which the conversion will be done (node, web, paw,
+  etc.). At this point, the user can set up a cache to help with the resolution of missing URIs.
+2. **Loading -** API-Flow tries to resolve a URI to a file and then parses this file to look for
+  missing references, and to fix/normalize parts of the file that can be improved
+3. **Parsing -** API-Flow parses the normalized file and converts it into its internal model
+4. **Serializing -** API-Flow converts the internal model representation into a target format.
 
-* [Importers]()
-* [Generators]()
-* [API Immutables]()
+If you want to see what a converter would look like, you can refer yourself to the file
+`src/api-flow.js`
 
+## Why use an internal model
+Because it's `n/2` times faster to implement than 1-to-1 conversions, where `n` is the number of
+format-versions that we want to support.
+
+## Decomposing the Internal Model
+The internal model tries to describe what an API can do with as reduced set of components that, in
+theory, should be able to describe an API as well as any API Format.
+
+### Core Components of the API-Flow model
+In this section we expose a small group of core components as well as some of their subtleties.
+This section does not exhaustively cover all components of the API-Flow, and leaves the more trivial
+ones to the comprehension of the reader.
+
+#### Api
+The Api Record is the root component that holds all the other components. It is composed of 4 fields
+that describe different aspects of an API.
+- **info -** the info field is represented by an Info Record that holds metadata relative to the
```

Comment:
Maybe hightlight more the Record names, either:

- `Info`
- **Info**
- *Info*

To make it clear it's a entity name.